extensions/extension.go

```diff
@@ -458,21 +463,21 @@ func loadFunc(env any, _ string, args []object.Object) object.Object {
 	file, err := sanitizeFileName(args)
 	s := env.(*eval.State)
 	if err != nil {
-		return s.Error(err.Error())
+		return s.ErrToError(err)
 	}
 	f, err := os.Open(file)
 	if err != nil {
-		return s.Error(err.Error())
+		return s.ErrToError(err)
 	}
 	defer f.Close()
 	all, err := io.ReadAll(f)
 	if err != nil {
-		return s.Error(err.Error())
+		return s.ErrToError(err)
 	}
 	// Eval the content.
 	res, err := eval.EvalString(env, string(all), false)
 	if err != nil {
-		return s.Error(err.Error())
+		return s.ErrToError(err)
```

Comment:
@ccoVeille I bet you probably don't like that name, suggestion for a better one?

Reply:
I do dislike it.

ReportError ?

ReturnError?

Reply:
overlord suggested a few, I picked ConvertError() but I could be convinced to use something else

not ReportError as it's not what it does... it makes an error object (which cause evaluation to abort)
```go
return s.ReturnError()
```
looks a bit stuttering :)

Reply:
maybe I can make the other one ErrorString() and use Error for this one

Reply:
I think this is actually decent now:
```go
/ NewError creates a new error object from a plain string.
// NewError will attach the stack trace to the NewError object.
func (s *State) NewError(msg string) object.Error
// Errorf formats an object.Error with the given format and args.
func (s *State) Errorf(format string, args ...interface{}) object.Error 
// Error converts from a go error to an object.Error.
func (s *State) Error(err error) object.Error 
```